src/modules/git_status.rs

```diff
@@ -290,30 +399,71 @@ struct RepoStatus {
     staged: usize,
     typechanged: usize,
     untracked: usize,
+    worktree_added: usize,
+    worktree_added_and_untracked: usize,
+    worktree_deleted: usize,
+    worktree_modified: usize,
+    worktree_typechanged: usize,
+    index_added: usize,
+    index_deleted: usize,
+    index_modified: usize,
+    index_typechanged: usize,
 }
 
 impl RepoStatus {
     fn is_deleted(short_status: &str) -> bool {
         // is_wt_deleted || is_index_deleted
-        short_status.contains('D')
+        Self::is_worktree_deleted(short_status) || Self::is_index_deleted(short_status)
```

Comment:
Please just add the counts as needed for the regular variables.

Reply:
Thanks for the review.

Can you clarify what you're asking for here, the existing counts where left un-touched:

https://github.com/starship/starship/blob/c8d74b101f81bb0c50615e00106ebabb4a0552ad/src/modules/git_status.rs#L468-L482

Are you asking for me to leave the logic as it was? Therefore reverting changes made to `is_deleted`, `is_modified`, `is_staged` and `is_typchanged`?

Or are you asking for something else? 

The changes where just an attempt to refactor, but will function the same as far as I understand.

Reply:
As far as I can tell, deleted and other variables are equal to e.g. `is_wt_deleted + is_index_deleted`, in that case I fell like it makes sense to keep track of these for each case and add these for the main `deleted` variable.

Reply:
Fixed in: https://github.com/starship/starship/pull/5655/commits/44b9dd8e75f85707e13860db376f3d098d7d85b1

Reply:
and then made clearer again here: https://github.com/starship/starship/pull/5655/commits/75b1e1b57ff02e5366b4175b002571f46dea2572